src/lib/components/composites/settings/project-settings/general/ProjectNameSettings.svelte

```diff
@@ -53,39 +55,46 @@
      * provided name.
      */
     async function handleSubmit(event: Event) {
-        updateProjectError = undefined;
-        loading = showLoadingSpinner = true;
         event.preventDefault();
-        const newProjectName: string = projectNameInput.getValue().trim();
-        const projectNameValid = projectNameInput.validate();
-        if (projectNameValid) {
-            const projectData: Partial<Project> = {
-                id: projectId,
-                name: newProjectName,
+        updateProjectError = undefined;
+
+        if (!projectNameInput.validate()) return;
+
+        const projectData: Partial<Project> = {
+            id: projectId,
+            name: projectNameInput.getValue(),
+        };
+
+        if (projectName === projectData.name) {
+            updateProjectError = {
+                errorTitle: "Please provide a different name.",
```

Comment:
What do you think about (based on this: [How to write error messages that actually help users rather than frustrate them](https://piccalil.li/blog/how-to-write-error-messages-that-actually-help-users-rather-than-frustrate-them/)):

```suggestion
                errorTitle: "Please provide a different name than the current name.",
```
Or something like this. Just different name doesn't tell the user why.

Reply:
I've updated it. Tell me what you think.

Reply:
Perfect, this is better!